### DISCUSSION TITLE
How to add custom widget in apostrophecms page ?

### DISCUSSION THREAD

**Discussion by sarathcm** [apostrophecms/apostrophe#3803] [Q&A] [answered]
Created a new widget based on apostrophecms documentation but did not get any proper idea of how to add a custom widget to the page. Is it possible to add a custom widget as a field option?

**Comment by boutell** [comment, marked as answer]
It is true that there is not a specific example of using the new widget on this page:

https://v3.docs.apostrophecms.org/guide/custom-widgets.html

But it works like adding any other widget:

```
add: {
  body: {
    type: 'area',
    label: 'Body',
    options: {
      widgets: {
        'two-column': {}
      }
    }
  }
}
```

This would add the `two-column-widget` module defined in the documentation you mention to an area, for instance an area in the home page or default page, or a piece type, or even another widget.

The rule is that you take `-widget` off the module name when adding a widget to the `widgets` section of an `area` field.
